app/routes.py

```diff
@@ -1,2 +1,260 @@
+
+    if task is None:
+        return make_response("", 404)
+
+    if request.method == "GET":
+        return jsonify({"task": task.display_tasks()}), 200
+
+    elif request.method == "PUT":
+        form_data = request.get_json()
+
+        task.title = form_data["title"]
+        task.description = form_data["description"]
+        task.completed_at = form_data["completed_at"]
+
+        db.session.commit()
+
+        return jsonify({"task": task.display_tasks()}), 200
+
+
+    elif request.method == "DELETE":
+        db.session.delete(task)
+        db.session.commit()
+        return jsonify({"details": (f'Task {task.id} "{task.title}" successfully deleted')}), 200
+    
+    
+    return {
+        "message": f"Task with id {task.id} was not found",
+        "success": False,
+    }, 404
+
+
+
+# MARK ID AS COMPLETE  
+@tasks_bp.route("/<task_id>/mark_complete", methods=["PATCH"], strict_slashes=False)
+def complete_task(task_id):
+
+    if request.method == "PATCH":
+        task = Task.query.get(task_id)
+        if task is None:
+            return jsonify(None), 404
+
+        task.completed_at = datetime.now()
+        db.session.commit()
+        slack_message(task)
+        return jsonify({"task": task.display_tasks()}), 200
+
+# SEND MESSAGE TO SLACK WHEN COMPLETE
+def slack_message(task):
+
+    slack_token = os.environ.get("TOKEN")
+    
+    url = "https://slack.com/api/chat.postMessage?channel=task-notifications"
+
+    payload = {"text": f"Someone just completed the task: '{task.title}'!"}
+    headers = {"Authorization": f"Bearer {slack_token}"}
+
+    return requests.request("PATCH", url, headers=headers, data=payload)
+
+
+# MARK ID AS INCOMPLETE 
+@tasks_bp.route("/<task_id>/mark_incomplete", methods=["PATCH"], strict_slashes=False)
+def incomplete_task(task_id):
+
+    if request.method == "PATCH":
+        task = Task.query.get(task_id)
+        if task is None:
+            return jsonify(None), 404
+
+        task.completed_at = None
+
+        return jsonify({"task": task.display_tasks()}), 200
+
+
+
+    
+# GET AND CREATE GOAL ITEMS
+@goals_bp.route("", methods=["GET", "POST"], strict_slashes=False)
+def one_goal():
+    
+    if request.method == "GET":
+        goals = Goal.query.all() # get all items in goal list
+        
+        goals_response = []
+
+        for goal in goals:
+            goals_response.append({
+            "id": goal.goal_id,
+            "title": goal.title
+            })
+
+        return jsonify(goals_response)
+
+    
+    if request.method == "POST":
+        request_body = request.get_json()
+        if ("title" not in request_body):
+            return make_response(jsonify({"details": "Invalid data"}), 400)
+        else:
+            goal = Goal(title = request_body["title"])
+        
+
+        db.session.add(goal)
+        db.session.commit()
+
+
+        return make_response({"goal": {
+            "id": goal.goal_id,
+            "title": goal.title
```

Comment:
💡 Refactor tip: notice how you return the same goal dictionary for GET and POST? I think a helper method to return a dictionary would come in handy here.